src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -317,14 +322,14 @@ private async Task ProcessGetDataAsync(INetworkPeer peer, GetDataPayload getData
             foreach (InventoryVector item in getDataPayload.Inventory.Where(inv => inv.Type.HasFlag(InventoryType.MSG_BLOCK)))
             {
                 // TODO: check if we need to add support for "not found"
-                Block block = await this.blockStore.GetBlockAsync(item.Hash).ConfigureAwait(false);
+                ChainedHeaderBlock chainedHeaderBlock = await this.consensusManager.GetBlockDataAsync(item.Hash).ConfigureAwait(false);
 
-                if (block != null)
+                if (chainedHeaderBlock.Block != null)
```

Comment:
chainedHeaderBlock can be null here in case peer asked for a bogus block. That will crash the node. Please add support for null as it says in the todo above

Reply:
Ahh good catch

Reply:
@zeptin please also fix this comment

Reply:
The todo above does not apply to missing blocks but rather to investigating if a response to the peer is needed.
So just check that also  `chainedHeaderBlock ` is not null probably just do
```
if (chainedHeaderBlock?.Block != null)

```